core/src/main/java/org/elasticsearch/index/query/support/InnerHitsQueryParserHelper.java

```diff
@@ -130,4 +132,20 @@ public static void parseCommonInnerHitOptions(XContentParser parser, XContentPar
             }
         }
     }
+
+    private static void parseHighlight(XContentParser parser, SearchContext context) {
+        QueryParseContext parseContext = context.getQueryShardContext( ).parseContext();
+        XContentParser old = parseContext.parser();
```

Comment:
InnerHitsQuerParserHelper is gone on master now, so this is not relevant here any more. I'm looking into the QueryParseContext simplifications however.